### ISSUE TITLE
Guard CMake Include of 

### ISSUE THREAD

**Issue by Kerstin-Keller** [eclipse-ecal/ecal#380]
Mon and Rec extension points are not part of eCAL core.
They should not be build with the regular settings.

https://github.com/continental/ecal/blob/152145cf5507833cbfb03e5d366a587c2a405a2e/CMakeLists.txt#L315-L321

They should be guarded with either `-DBUILD_APPS` or other options: `BUILD_REC_SDK`, `BUILD_MON_SDK` ???

@FlorianReimold @hannemn

**Comment by FlorianReimold** [comment]
Yes, I agree. They don't have additional dependencies, but they should probably only be built when apps are built.
